### ISSUE TITLE
Small inaccuracies with which data is shown and Japanese icons

### ISSUE THREAD

**Issue by HilbertIdeals5** [RiiConnect24/File-Maker#50]
Short Description of Issue: 
These are two things with the current Forecast Channel setup that make it inaccurate compared to footage from before the shutdown, those being that 1) data for past hours is shown and 2) the 6-hourly intervals for Japanese cities show two icons instead of one.

Long Issue: 
Note that my input here will be of dubious authenticity because I'm basing this off videos and not actual experience, so you may take this with a grain of salt.
Here are the two issues:
1. Data for past hours
In the video I have linked below, numbers for the chance of rain "Today" are replaced by dashes for the current 6-hour quarter and the previous ones.
<img width="320" alt="Screen Shot 2021-02-05 at 2 02 04 PM" src="https://user-images.githubusercontent.com/63978875/107152779-c722da80-6937-11eb-95d5-b0b22ed157b7.png">
^^ It's 11, so no percentage amounts of rain are shown for 0-6 and 6-12, but are shown for 12-18 and 18-24.
It's also the same for the 6-hourly icons. I'm not sure why the low temperature has dashes though.<br>On the other hand, with RiiConnect24, all 4 intervals have weather data, even ones in the past.<br><br>

2. Japanese weather icons
Based on a _lot_ of research ~~to feed my addiction for the Forecast Channel~~, I have an idea of what the weather icons are for. One major thing that I found is that for hourly weather, only one icon is used, not two.
Instead of this:
![rc24-fc jpsymb](https://user-images.githubusercontent.com/63978875/107152932-d3f3fe00-6938-11eb-9454-c45850583405.png)
It's just this:
<img width="320" alt="Screen Shot 2021-02-05 at 2 03 02 PM" src="https://user-images.githubusercontent.com/63978875/107152985-259c8880-6939-11eb-91ce-e70dcacb86bc.png">
In fact, there is actually a system to which icons are used:

1. if there's generally only one type of weather throughout the day, or for the hourly weather, only one icon is used, and it's called just "clear/cloudy/rainy/snowy/stormy"
2. if one type is predominant but the day occasionally includes a secondary type, then two are used, with the primary one being the upper one, and it's called "-- with some --"
3. if there are two types of weather, with one at the start of the day and the other at the end, then the combination with the arrow at the bottom is used, and it's called "-- into --"
<img width="320" alt="Screen Shot 2021-02-05 at 2 03 46 PM" src="https://user-images.githubusercontent.com/63978875/107153214-49140300-693a-11eb-8672-42197c6abe68.png">

This is also evident on the Weathernews site: http://weathernews.jp/onebox/40.714270/-74.005970/. Since RC24 uses AccuWeather as a source and because I'm not Japanese, I'm not too worried about the icons being used correctly, but it might be important to consider.

Here is the video I used for reference:
https://youtu.be/KRnLPHXUc_A

Steps to Reproduce :
- Go to the Forecast Channel from the Wii Menu, then go to the Globe
- Click on any city, and there are icons for 6-hourly intervals that might be in the past
- Click on any Japanese city, and you might find the combination icon in a 6-hourly interval

**Comment by larsenv** [comment]
Interesting. Are you saying that Nintendo/Weathernews used different icons for the weather? Since we use AccuWeather, they don't really show conditions for 2 types of weather in different parts of the day.

What we do for AccuWeather is take the weather icon that the API gives us and we use that. I guess Weathernews did it differently.

The precipitation issue is likely fixed now that I fixed the hourly data as mentioned in your other issue, thanks for that.

**Comment by HilbertIdeals5** [comment]
**Japanese Icons:**

For the Japanese icons, this is based on what I saw after looking at the Weathernews website and other resources. What I saw is not really that there were only two types of weather at different times of day. It's just that the hourly weather is only expressed with a single icon, and the overall daily weather is expressed with two icons.

So:
1. for hourly weather/current weather, only one icon is used no matter what.
2. if there's only one "type" of weather throughout the day, only one icon is used. For example, if it's cloudy throughout the majority of the day, the icon for the day would be "☁️", and the text underneath would say "Cloudy".
3. if the weather varies throughout the day, with one type for most of the day and another type occurring occasionally, then two icons are used. For example, if the day will be clear but occasionally cloudy, then the icon for the day would be "☀️☔️", and the text underneath would say "Sunny with some Rain".
4. if the weather transitions from being one type in the morning to another type in the evening, then two icons with an arrow beneath them is used. For example, if the day will be cloudy in the morning and sunny in the evening, then the icon for the day would be "☁️>☀️", and the text underneath would say "Cloudy into Sunny".

You might want to look at a couple of forecasts for different cities using the Weathernews link I provided to get a good idea of what I mean.

Based on the interviews with the developers of the Forecast Channel (https://interviews.rc24.xyz/story_forecast.html), I assume this system is used because the weather in Japan is not as varied. In some places, the weather could change multiple times within the hour, which is why the complex icons we're familiar with made more sense than the Japanese ones. They also expressed confusion over the thunderstorm icon.

Also, just in case to make things clear, the weather icons for places other than Japan are fine.

**Precipitation/Weather:**

![Screen Shot 2021-02-13 at 4 54 24 PM](https://user-images.githubusercontent.com/63978875/107863273-f6968300-6e20-11eb-9634-4b0d87966331.png)

This issue has been fixed, but only partially. I took this screenshot at 4:00 pm, so I was expecting that only the last section for 6pm-12am would have an icon, and the other three sections would have "--".

Otherwise, the sun and moon are in the correct spots:
![Screen Shot 2021-02-13 at 8 52 24 PM](https://user-images.githubusercontent.com/63978875/107866325-877b5780-6e3d-11eb-8be0-a604c8302f8c.png)

**Comment by larsenv** [comment]
I see you've done your homework.

I'm not interested in using Weathernews for our weather conditions as AccuWeather works perfectly for our use. I can send about 10k API calls in about 3 minutes! Weathernews a weather source that's mostly used in Japan, and doesn't have a userbase overseas.

I think it's cool how Weathernews did it differently in Japan, but I don't see a reason why this is a big deal. I never really thought about it like you have.

The hourly Forecast is correct, AccuWeather just returns the hourly Forecast for the whole day I assume?

Yeah, today I had to modify it. Turns out the Forecast was displaying incorrectly if Today Forecast is for a country that already has progressed to the next day. For example, if today's February 13th and it is already February 14th in Australia.

**Comment by HilbertIdeals5** [comment]
If using the Japanese icon system isn't a concern, then I understand. Also, I don't mean to suggest you should switch to Weathernews, since you've already set this system up around AccuWeather.

Thanks for going through all of this, I didn't expect you'd follow along this much 😊

**Comment by HilbertIdeals5** [comment]
Another thing though......
![Screen Shot 2021-02-14 at 2 51 23 PM](https://user-images.githubusercontent.com/63978875/107887494-8cd7b100-6ed4-11eb-842e-98c48cfcddc2.png)
Sun in the 6pm-12am section, I only saw this once so maybe it's a small glitch?

EDIT 2/15: I also found this:
![Screen Shot 2021-02-15 at 2 35 25 PM](https://user-images.githubusercontent.com/63978875/107987945-fa4f1480-6f9c-11eb-8947-8148d49dae06.png)
Based on the AccuWeather website, maybe it's because the sun icon is associated with "Cold", and "Partly sunny" (at 6PM exactly) is used for the 6pm-12am section? I saw the same thing for Sioux Falls, SD.

EDIT 2/22: After checking for a few days, it still persists:
<img width="1440" alt="Screen Shot 2021-02-22 at 10 44 48 AM" src="https://user-images.githubusercontent.com/63978875/108732448-9330e300-74fb-11eb-9683-a01538b704b5.png">
